veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,253 @@
+
+However, container runtimes drop all capabilities from the [bounding set](https://man7.org/linux/man-pages/man7/capabilities.7.html), and capabilities must explicitly be added in the SecurityContext, as done for CAP_NET_BIND_SERVICE in [virt-launcher](https://github.com/kubevirt/kubevirt/blob/release-1.5/pkg/virt-controller/services/rendercontainer.go#L284).
+
+Since virt-launcher drops the root user, this method is secure. However, KubeVirt follows Kubernetes' `Restricted` Pod Security Standard, which disallows CAP_NET_ADMIN.
+
+## Scalability
+
+- `passt` has a memory overhead of ~250Mi per VM. Users running VMs at scale, should revert to other bindings if memory overhead is a concern.
+-  VMs with a large number of TCP connections, will take longer to live migrate. That's because sockets are handled in a serial
+ fashion, each one separately sent back and forth to passt-repair for handling.
+- The number of concurrent live migration per node is already limited and bound by the number of virt-handler worker goroutines.
+
+## Update/Rollback Compatibility
+
+-Live migration of passt VMs from nodes running older versions of Kubevirt to an upgraded Kubevirt environment will cause a dis
+connection of TCP connections.
+-Rollback of passt migration functionality (in case system stability is impacted) will be possible by disablement of the feature
+e gate. It is expected that passt would still work, but live migrations would disconnect TCP connections.
+
+
+## Functional Testing
+
+Existing e2e tests cover basic passt connectivity and remain sufficient with vhost-user support. Dedicated e2e tests for seamless migration will verify TCP connection persistence. Sticky IPs are required, so tests will use secondary interfaces with hardcoded IPs.
+
+## Implementation Phases
+1. Implement vhost-user in passt-sidecar.
+2. Add passt-repair to virt-handler.
+3. Implement calls to passt-repair in virt-handler.
+4. Enhance e2e tests.
+5. Deployment activities (omitted from this proposal).
+
+Phases can proceed in parallel.
+
+## Feature Lifecycle Phases
+
+`passt` is currently a network binding plugin external to KubeVirt core, and its use as a network binding is not subject to lifecycle phases. However, the calls to passt-repair from virt-handler impose a risk and should be protected by a feature gate and subject to lifecycle phases.
+
+### Alpha
+
+`passt live migration` will be feature-gated, requiring explicit user opt-in. virt-handler will not run `passt-repair` unless the feature gate is enabled.
+
+passt-sidecar will not condition the feature gate and will populate DomainXML in vhost-user mode regardless of the gate. This part of the code is external to KubeVirt core.
+
+### Beta
+
+For several releases, the live migration process using `passt` will be optimized. Specifically, `past-repair` and its integration points will be ironed out. 
```

Comment:
`pasts-repair`

Reply:
Sorry, I just realised, there's a typo in my comment about your typo (that's what my comment was about). `passt-repair`, not `past-repair`. :)